### PR TITLE
Add support for SpeechBrain SoundChoice G2P (speechbrain part)

### PR DESCRIPTION
**Pull request by flexthink**
Hosted Interface API for the SoundChoice G2P model (speechbrain part)

https://huggingface.co/speechbrain/soundchoice-g2p
https://arxiv.org/abs/2207.13703

### CODE REVIEW

**Comment by flexthink** [comment]
This needs to be merged first (otherwise the grapheme-to-phoneme task will not be allowed by validation):
https://github.com/huggingface/api-inference-community/pull/100

**Comment by osanseviero** [comment]
Copy-paste of what was mentioned in #100 

Hey there! For simplicity and given this is the first model on the Hub for this task, instead of creating a new type `grapheme-to-phoneme`, maybe we can reuse `text2text-generation`. This has some benefits:

* It has same input and output types (text to text)
* The widget and API would work out of the box

This would be easier to maintain and we could launch it very quickly. One last comment, instead of joining with a space, maybe we can use a hyphen (-) so the output string is like "T-UW- -B-IY" as some phonemes can be empty strings

```python
from speechbrain.pretrained import GraphemeToPhoneme
g2p = GraphemeToPhoneme.from_hparams("speechbrain/soundchoice-g2p")
text = "To be or not to be, that is the question"
phonemes = g2p(text)
# ['T', 'UW', ' ', 'B', 'IY', ' ', 'AO', 'R', ' ', 'N', 'AA', 'T', ' ', 'T', 'UW', ' ', 'B', 'IY', ' ', 'DH', 'AE', 'T', ' ', 'IH', 'Z', ' ', 'DH', 'AH', ' ', 'K', 'W', 'EH', 'S', 'CH', 'AH', 'N']
```

File: docker_images/speechbrain/app/pipelines/text2text_generation.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        """
+        Args:
+            inputs (:obj:`str`):
+                The input text
+        Return:
+            A :obj:`dict` A dictionary: {"text": the model output)}
+        """
+        output = self.model(inputs)
+        output = self.post_process(output)
+        return {"text": output}
```
[the remark this answers is not in the document]
Comment (flexthink) [reply docker_images/speechbrain/app/pipelines/text2text_generation.py:28]: @osanseviero : I have no way of telling:
* There are no other implementations of `text2text-generation`
* Other tasks with text output have a key called `text` (e.g. ASR). 
* I tried a search of all `huggingface` repositories and didn't find anything for `generated_text`

[the remark this answers is not in the document]
Comment (flexthink) [reply docker_images/speechbrain/app/pipelines/text2text_generation.py:28]: @osanseviero: Please tell me which key I should use, and I will update accordingly

[the remark this answers is not in the document]
Comment (osanseviero) [reply docker_images/speechbrain/app/pipelines/text2text_generation.py:28]: Yes, the key should be `generated_text` as that's what's used in the [widget](https://github.com/huggingface/hub-docs/blob/main/js/src/lib/components/InferenceWidget/widgets/TextGenerationWidget/TextGenerationWidget.svelte#L158) and the outputs of the API as per [the docs](https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task)

[the remark this answers is not in the document]
Comment (flexthink) [reply docker_images/speechbrain/app/pipelines/text2text_generation.py]: @osanseviero : I pushed the change

**Comment by flexthink** [comment]
@osanseviero : Is anything else required to enable the widget? I tried to enable it and got the error below.

image

Here is the model page:
https://huggingface.co/speechbrain/soundchoice-g2p

**Comment by osanseviero** [comment]
The docker container still needs to be deployed (cc @Narsil), you can follow the progress at https://github.com/huggingface/api-inference-community/issues/102

**Pull request merged**